### PR TITLE
[cli] Improve span merging in the coverage tool

### PR DESCRIPTION
**Pull request by Rqnsom**
## Description
- Improve merging spans in the coverage tool
  - The old solution failed to merge neighbouring spans: e.g. `[3, 5]` and `[5, 7]` into `[3, 7]`
- Also, make the `merge_spans` function public so it could be used by the `move-mutation-test` tool that also parses the coverage output file.

## How Has This Been Tested?
- A new unit test has been written.
- `cargo clippy` hasn't introduced any new warnings for the new change.
- I ran the `aptos move coverage` tool locally to ensure it still works as expected.

## Key Areas to Review
This should have no visible impact on current `aptos move coverage` tool. The old limitation was not an issue, since displaying covered lines with always worked fine although spans were not merged "perfectly". This improvement is much more important for the usage in the `move-mutation-test` tool.

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [x] Performance improvement
- [x] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [x] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Move Compiler
- [ ] Other (specify)

## Checklist
- [x] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [x] I tested both happy and unhappy path of the functionality
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by vineethk — COMMENTED** [review]
Minor comments, will approve once fixed.

File: third_party/move/tools/move-coverage/src/source_coverage.rs [hunk 1]
```diff
@@ -398,34 +398,106 @@ impl SourceCoverage {
...
+
+            // Should be merged:
+            Loc::new(test_hash, 20, 25),
+            Loc::new(test_hash, 24, 25),
+            Loc::new(test_hash, 21, 25),
+
+            // Should stay the same:
+            Loc::new(test_hash, 26, 29),
+
+            // Shouldn't intefer with other hashes.
```
[the remark this answers is not in the document]
Comment (Rqnsom) [reply third_party/move/tools/move-coverage/src/source_coverage.rs]: Done

File: third_party/move/tools/move-coverage/src/source_coverage.rs [hunk 2]
```diff
@@ -398,34 +398,106 @@ impl SourceCoverage {
...
-            curr = curr.merge(interval);
+    for span in spans {
+        if curr.end() >= span.start() {
+            curr = curr.merge(span);
+            continue;
         }
+
+        unioned.push(curr);
+        curr = span;
     }
```
[the remark this answers is not in the document]
Comment (Rqnsom) [reply third_party/move/tools/move-coverage/src/source_coverage.rs]: Done

**Review by vineethk — APPROVED** [review]
Thanks for addressing the comments!

**Review by brmataptos — APPROVED** [review]
Looks ok, though the coverage problems are deeper than this.

**Pull request merged**